File: backend/src/main/java/com/example/demo/service/StatsService.java
// src/main/java/com/example/demo/service/StatsService.java
package com.example.demo.service;

import com.example.demo.dto.stats.RevenueDataPointDTO;
import com.example.demo.dto.stats.StatsDTO;

import java.time.LocalDate;
import java.util.List;


public interface StatsService {


    StatsDTO getBasicStats(LocalDate startDate, LocalDate endDate);

    StatsDTO getTodayStats();

    StatsDTO getLast7DaysStats();

    StatsDTO getCurrentMonthStats();

    List<RevenueDataPointDTO> getRevenueOverTime(LocalDate startDate, LocalDate endDate);

}

File: frontend/src/http/modules/public/statsService.js
// src/http/statsService.js
import apiClient from "@/http/axios.js"; // Import apiClient đã cấu hình (tự động gửi token)

// Lấy thống kê hôm nay
export const getTodayStats = () => {
  return apiClient.get("/admin/stats/today"); // API Backend đã tạo
};

// Lấy thống kê 7 ngày qua
export const getLast7DaysStats = () => {
  return apiClient.get("/admin/stats/last7days"); // API Backend đã tạo
};

// Lấy thống kê tháng hiện tại
export const getCurrentMonthStats = () => {
  return apiClient.get("/admin/stats/current-month"); // API Backend đã tạo
};

// Lấy thống kê theo khoảng ngày tùy chọn
// startDate và endDate phải là string dạng 'YYYY-MM-DD'
export const getBasicStats = (startDate, endDate) => {
  return apiClient.get("/admin/stats/basic", {
    params: {
      startDate, // ví dụ: '2024-01-01'
      endDate, // ví dụ: '2024-03-31'
    },
  });
};
// --- HÀM MỚI ---
// Lấy dữ liệu doanh thu theo ngày cho biểu đồ
export const getRevenueOverTime = (startDate, endDate) => {
  return apiClient.get("/admin/stats/revenue-over-time", {
    params: {
      startDate, // 'YYYY-MM-DD'
      endDate, // 'YYYY-MM-DD'
    },
  });
};
